app/src/main/java/com/framgia/forder/data/source/remote/ProductRemoteDataSource.java

```diff
@@ -32,4 +37,54 @@ public ProductRemoteDataSource(FOrderApi api) {
                     }
                 });
     }
+
+    @Override
+    public Observable<List<Product>> getListProductInShop(int shopId, int domainId) {
+        return mFOrderApi.getListProductShop(shopId, domainId)
+                .flatMap(new Func1<ProductResponse, Observable<List<Product>>>() {
+                    @Override
+                    public Observable<List<Product>> call(ProductResponse productResponse) {
+                        if (productResponse != null) {
+                            return Observable.just(productResponse.getListProduct());
+                        }
+                        return Observable.error(new NullPointerException());
+                    }
+                });
+    }
+
+    @Override
+    public Observable<Void> orderProduct(OrderRequest orderRequest) {
+        return mFOrderApi.orderRequest(orderRequest)
+                .flatMap(new Func1<OrderResponse, Observable<Void>>() {
+                    @Override
+                    public Observable<Void> call(OrderResponse orderResponse) {
+                        if (orderResponse != null) {
+                            return Observable.just(null);
+                        }
+                        return Observable.error(new NullPointerException());
+                    }
+                });
+    }
+
+    @Override
+    public Observable<List<Comment>> getListCommentInProduct(int productId, int domainId) {
+        List<Comment> comments = new ArrayList<>();
+        comments.add(new Comment(1, 1, "Trần Đức Quốc", "Rất ngon", "26/04/2017"));
```

Comment:
WARNING: [AvoidDuplicateLiterals] The String literal "Rất ngon" appears 9 times in this file; the first occurrence is on line 72
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/strings.html#AvoidDuplicateLiterals